packages/app/src/config/chain/index.ts

```diff
@@ -140,19 +143,57 @@ const chainConfig: ChainConfig = {
         type: 'yielding-fixed',
         baseAsset: TokenSymbol('ETH'),
         providedBy: ['chainlink'],
-        ratio: async () => NormalizedUnitNumber(0),
+        ratio: async ({ reserve, wagmiConfig }) => {
+          const stethAddress = await readContract(wagmiConfig, {
+            abi: wstEthOracleAbi,
+            address: reserve.priceOracle,
+            functionName: 'steth',
+            args: [],
+            chainId: mainnet.id,
+          })
+
+          const ratio = await readContract(wagmiConfig, {
+            abi: wstEthRatioAbi,
+            address: stethAddress,
+            functionName: 'getPooledEthByShares',
+            args: [toBigInt(10 ** reserve.token.decimals)],
+            chainId: mainnet.id,
+          })
+
+          return NormalizedUnitNumber(BaseUnitNumber(ratio).dividedBy(10 ** reserve.token.decimals))
+        },
       },
       [TokenSymbol('rETH')]: {
         type: 'yielding-fixed',
         baseAsset: TokenSymbol('ETH'),
         providedBy: ['chainlink'],
-        ratio: async () => NormalizedUnitNumber(0),
+        ratio: async ({ reserve, wagmiConfig }) => {
+          const ratio = await readContract(wagmiConfig, {
+            abi: rEthRatioAbi,
+            address: reserve.token.address,
+            functionName: 'getExchangeRate',
+            args: [],
+            chainId: mainnet.id,
+          })
+
+          return NormalizedUnitNumber(BaseUnitNumber(ratio).dividedBy(10 ** reserve.token.decimals))
+        },
       },
       [TokenSymbol('weETH')]: {
         type: 'yielding-fixed',
         baseAsset: TokenSymbol('ETH'),
         providedBy: ['chainlink'],
-        ratio: async () => NormalizedUnitNumber(0),
+        ratio: async ({ reserve, wagmiConfig }) => {
+          const ratio = await readContract(wagmiConfig, {
+            abi: weEthRatioAbi,
+            address: reserve.token.address,
+            functionName: 'getRate',
+            args: [],
+            chainId: mainnet.id,
+          })
+
+          return NormalizedUnitNumber(BaseUnitNumber(ratio).dividedBy(10 ** reserve.token.decimals))
```

Comment:
nit: 
```suggestion
          return NormalizedUnitNumber(BaseUnitNumber(ratio).shiftedBy(-reserve.token.decimals))
```
is more canonical in our codebase

Reply:
Isn't `token.fromBaseUnit` the way to go 😬